crates/rattler_solve/src/lib.rs

```diff
@@ -70,6 +70,22 @@ impl fmt::Display for SolveError {
     }
 }
 
+/// Represents the channel priority option to use during solves.
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub enum ChannelPriority {
+    /// The channel that the package is first found in will be used as the only channel
+    /// for that package.
+    Strict,
+
+    /// Conda also has "Flexible" as an option, where packages present in multiple channels
+    /// are only taken from lower-priority channels when this prevents unsatisfiable environment
+    /// errors, but this would need implementation in the solvers.
+    /// Flexible,
+
+    /// Packages can be retrieved from any channel as package version takes precedence.
+    Disabled,
```

Comment:
Shoudl we call this `AllEqual` or `NoPriority/NoPriorities`? Just thinking of something that makes it really clear.

Reply:
Yeah, totally. I am also OK with keeping the naming. Just sometimes I think we have the opportunity to come up with new & improved terminology :)